Converting e2e tests into Integration tests for Kiali, initial commit with Kiali Status test for CI. (#4868)
kiali/kiali: tests/integration/README.md

# Kiali Integration Tests

### Project Structure

* `tests/integration/`:  Root directory
    *  `tests`:  Tests
    *  `utils`:  Support utilities

## Environment Setup

* System dependencies that will need to be available
    *  `oc`
    *  `go`
    *  `make`
    *  `npm`
    *  `yarn`

> * It is expected that the following have already been deployed in Openshift

Reviewer: Is there a reason this README is geared towards OpenShift? I do not see instructions here on how to run these integration tests on non-OpenShift environments (like minikube). We should have the ability to run these i-tests on an OpenShift and a non-OpenShift cluster - instructions on how to do both are needed.

Author: so far oc is not required, will remove it for now

    * `Istio-system (Istio, Grafana, Prometheus, etc.)`
    * `Kiali (Deployed into Istio-system namespace)`
    * `Bookinfo (with traffic generated to Bookinfo)`

```sh
# Clone this repository
$ git clone https://github.com/kiali/kiali.git

# Install requirements
$ make test-integration-setup

# Login to Openshift Cluster 
$ oc login https://<openshift>:<port> --username=<username> --password=<password> --insecure-skip-tls-verify=true

> # Read the Token from Openshift Cluster, to be used in next step or running tests
> $ oc whoami -t
> 
> # run all tests
> $ make test-integration URL="https://<kiali-hostname>" TOKEN="<token>"

Reviewer: you can combine these so the user doesn't have to manually cut-n-paste the token to pass to the TOKEN env var:
```suggestion
# run all tests authenticating as the user that was logged in
$ make test-integration URL="https://<kiali-hostname>" TOKEN="$(oc whoami -t)"
```
